app/src/main/java/com/duckduckgo/app/browser/logindetection/NavigationAwareLoginDetector.kt

```diff
@@ -65,31 +79,91 @@ class NextPageLoginDetection @Inject constructor() : NavigationAwareLoginDetecto
             is NavigationEvent.UserAction -> {
                 discardLoginAttempt()
             }
+            is NavigationEvent.Redirect -> {
+                loginDetectionJob?.cancel()
+
+                val validUrl = Uri.parse(navigationEvent.url).getValidUrl() ?: return
+                if (validUrl.isOAuthUrl() || validUrl.isSSOUrl() || authDetectedHosts.any { validUrl.baseHost.contains(it) }) {
+                    authDetectedHosts.add(validUrl.baseHost)
+                    Timber.d("LoginDetectionDelegate Auth domain added $authDetectedHosts")
+                }
+
+                if (loginAttempt != null) {
+                    urlToCheck = navigationEvent.url
+                }
+                return
+            }
             is NavigationEvent.GpcRedirect -> {
                 gpcRefreshed = true
             }
         }
     }
 
+    private fun scheduleLoginDetection(): Job {
+        return GlobalScope.launch(Dispatchers.Main) {
```

Comment:
Good call on this. I will take a look and see how easy/complex is pass by a constructor argument another scope with our DI setup. Will check 👍

Reply:
After exploring some time how to address this comment, I couldn't find a proper way to introduce a scope here.

Basically, `NavigationAwareLoginDetector` is injected into `ViewModelFactory`, and that's the instance we use to create `BrowserTabViewModel` via constructor.
To have the `BrowserTabViewModel` scope inside `NavigationAwareLoginDetector`, I need to create the ViewModel first in order to get the scope and pass it to the login detector, which seems not possible, or at least I couldn't find how to do it.
I guess another option is to create an `init` method inside `NavigationAwareLoginDetector` to provide the scope after  `BrowserTabViewModel`, but it feels weird to me. 

Thoughts on that?

Reply:
I've just tried another approach: to create a single module for the login detection, so we could get an activity context / lifecycle and control the coroutine scope with that. However, our setup is a constraint for that approach. Since the ViewModelFactory is global, we would need to create another one, and refactor will be bigger than expected.

I'm going to leave a comment there pointing out that `globalScope` is used in that case because currently we can't easily inject a more concrete scope or a lifecycleScope.

Reply:
That's all fine, and I figured that a scope more closely aligned with the actual tab would be tricky so that's not surprising to hear.

One last thought: what about if `NavigationAwareLoginDetector` had a scope of its own instead of `GlobalScope`, in the same way that there is a `@AppCoroutineScope applicationCoroutineScope: CoroutineScope` defined?

Two things to highlight:

1. I don't know that this is really much of an improvement, so just asking the question - but not requesting it be done
2. It doesn't need to block this PR. Feel free to merge and carry on discussion async/elsewhere.

Reply:
I thought about the same but discarded using a scope that is equivalent to `GlobalScope` because:
- that scope is not tied to real consumer lifecycle
- It will "hide" the issue you spotted. Having a concrete scope injected will makes us think that the injected scope is the one needed there, but in the case of `applicationCoroutineScope` is not the case.